app/services/SalesforceService.scala

```diff
@@ -36,15 +49,29 @@ object SalesforceService {
     Keys.ALLOW_GU_RELATED_MAIL -> personalData.receiveGnmMarketing
   ) ++ personalData.title.fold(Json.obj())(title => Json.obj(
     Keys.TITLE -> title.title
-  )) ++ paperData.map(_.deliveryAddress).fold(Json.obj())(addr => Json.obj(
+  )) ++ paperData.map(_.deliveryRecipient).filterNot(_.isGiftee).map(_.address).fold(Json.obj())(addr => Json.obj(
```

Comment:
This stops us upserting the delivery recipient's address into the buyer's delivery address if they are buying a gift.

Reply:
this line looks very similar to  [line 218 in CheckoutService](https://github.com/guardian/subscriptions-frontend/pull/1222/files#diff-68b2eee9ecf473f772caa4b38dbdaa17R218). 
if we could extract it to a common location we could remove the duplication.  we could call it correspondence address and put it DeliveryRecipient, but I don't know if conceptually it would make sense here to use the "correspondence address"

Reply:
I have an idea on how to improve this. I'll submit that in an additional PR later today.